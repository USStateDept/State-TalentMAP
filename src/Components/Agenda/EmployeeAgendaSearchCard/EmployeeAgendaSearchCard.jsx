import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FA from 'react-fontawesome';
import { Tooltip } from 'react-tippy';
import { Handshake } from 'Components/Ribbon';
import LinkButton from 'Components/LinkButton';
import { get } from 'lodash';
import { format, isDate } from 'date-fns-v2';
import BoxShadow from '../../BoxShadow';

const EmployeeAgendaSearchCard = ({ isCDO, result }) => {
  // will need to update during integration
  const { person, currentAssignment } = result;
  const agendaStatus = get(result, 'agendaStatus') || 'Coming soon';
  const author = get(result, 'author') || 'Coming soon';
  const bidder = get(person, 'fullName') || 'None listed';
  const cdo = get(result, 'cdo') || 'Coming soon';
  const currentPost = get(currentAssignment, 'orgDescription') || 'None listed';
  const futurePost = get(result, 'futurePost') || 'Coming soon';
  const panelDate = get(result, 'panelDate') || 'Coming soon';
  const showHandshakeIcon = get(result, 'hs_accepted') || false;
  const ted = get(currentAssignment, 'TED') || '';
  const perdet = get(person, 'perdet', '');
  const userRole = isCDO ? 'cdo' : 'ao';
  const hideCreate = true;
  const employeeID = get(person, 'employeeID', '');

  const formatDate = (d) => isDate(new Date(d)) ? format(new Date(d), 'MM/yy') : 'None listed';

  return (
    <BoxShadow className="employee-agenda-stat-card">
      <div className="employee-agenda-card-inner">
        <div className="employee-agenda-card-top">
          <div className="employee-ribbon-container">
            <div className="ribbon-container-condensed">
              {showHandshakeIcon &&
                  <Tooltip
                    title="Handshake"
                    arrow
                    offset={-60}
                  >
                    <Handshake showText={false} className="ribbon-condensed-card" />
                  </Tooltip>
              }
            </div>
          </div>
        </div>
        <div>
          <h3>
            { isCDO ? <Link to={`/profile/public/${perdet}`}>{bidder}</Link> : bidder }
          </h3>
        </div>
        <div className="employee-agenda-card-data-point-top">
          <div className="employee-card-data-point">
            <FA name="id-badge" />
            <dt>ID:</dt>
            <dd>{employeeID}</dd>
          </div>
          <div className="employee-card-data-point">
            <FA name="building-o" />
            <dt>Org:</dt>
            <dd>
              {currentPost}
              <FA className="org-fa-arrow" name="long-arrow-right" />
              {futurePost}
            </dd>
          </div>
          <div className="employee-card-data-point">
            <FA name="clock-o" />
            <dt>TED:</dt>
            <dd>{ted ? formatDate(ted) : 'None listed'}</dd>
          </div>
          <div className="employee-card-data-point">
            <FA name="user-o" />
            <dt>CDO:</dt>
            <dd>{cdo}</dd>
          </div>
        </div>
        <div className="employee-card-data-point">
          <FA name="pencil-square" />
          <dt>Author:</dt>
          <dd>{author}</dd>
        </div>
        <div className="employee-card-data-point">
          <FA name="calendar-o" />
          <dt>Panel Meeting Date:</dt>
          <dd>{panelDate}</dd>
        </div>
        <div className="employee-card-data-point">
          <FA name="sticky-note-o" />
          <dt>Agenda Status:</dt>
          <dd>{agendaStatus}</dd>
        </div>
      </div>
      <div className="employee-agenda-card-bottom">
        <div className="button-container">
          <div className="view-agenda-item-container">
            <LinkButton className="view-agenda-item-button" toLink={`/profile/${userRole}/agendaitemhistory/${perdet}`}>View History</LinkButton>
          </div>
          <div className="create-ai-box-container">
            {!hideCreate &&
              <LinkButton className="create-ai-box-button" toLink="#">Create Agenda Item</LinkButton>
            }
          </div>
        </div>
      </div>
    </BoxShadow>
  );
};

EmployeeAgendaSearchCard.propTypes = {
  isCDO: PropTypes.bool,
  result: PropTypes.PropTypes.shape({ person: {}, currentAssignment: {} }),
};

EmployeeAgendaSearchCard.defaultProps = {
  isCDO: false,
  result: {},
};

export default EmployeeAgendaSearchCard;
