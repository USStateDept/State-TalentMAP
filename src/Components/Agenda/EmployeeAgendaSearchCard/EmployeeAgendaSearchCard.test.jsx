import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import EmployeeAgendaSearchCard from './EmployeeAgendaSearchCard';

describe('EmployeeAgendaSearchCards', () => {
  it('is defined', () => {
    const wrapper = shallow(<EmployeeAgendaSearchCard />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<EmployeeAgendaSearchCard />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
