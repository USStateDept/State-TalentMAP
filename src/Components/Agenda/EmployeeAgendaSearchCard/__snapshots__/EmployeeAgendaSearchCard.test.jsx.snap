// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeAgendaSearchCards matches snapshot 1`] = `
<BoxShadow
  blurRadius={10}
  className="employee-agenda-stat-card"
  color="rgba(0,0,0,.15)"
  inset={false}
  is="div"
  offsetX={3}
  offsetY={2}
  spreadRadius={1}
  style={Object {}}
>
  <div
    className="employee-agenda-card-inner"
  >
    <div
      className="employee-agenda-card-top"
    >
      <div
        className="employee-ribbon-container"
      >
        <div
          className="ribbon-container-condensed"
        />
      </div>
    </div>
    <div>
      <h3>
        None listed
      </h3>
    </div>
    <div
      className="employee-agenda-card-data-point-top"
    >
      <div
        className="employee-card-data-point"
      >
        <FontAwesome
          name="id-badge"
        />
        <dt>
          ID:
        </dt>
        <dd />
      </div>
      <div
        className="employee-card-data-point"
      >
        <FontAwesome
          name="building-o"
        />
        <dt>
          Org:
        </dt>
        <dd>
          None listed
          <FontAwesome
            className="org-fa-arrow"
            name="long-arrow-right"
          />
          Coming soon
        </dd>
      </div>
      <div
        className="employee-card-data-point"
      >
        <FontAwesome
          name="clock-o"
        />
        <dt>
          TED:
        </dt>
        <dd>
          None listed
        </dd>
      </div>
      <div
        className="employee-card-data-point"
      >
        <FontAwesome
          name="user-o"
        />
        <dt>
          CDO:
        </dt>
        <dd>
          Coming soon
        </dd>
      </div>
    </div>
    <div
      className="employee-card-data-point"
    >
      <FontAwesome
        name="pencil-square"
      />
      <dt>
        Author:
      </dt>
      <dd>
        Coming soon
      </dd>
    </div>
    <div
      className="employee-card-data-point"
    >
      <FontAwesome
        name="calendar-o"
      />
      <dt>
        Panel Meeting Date:
      </dt>
      <dd>
        Coming soon
      </dd>
    </div>
    <div
      className="employee-card-data-point"
    >
      <FontAwesome
        name="sticky-note-o"
      />
      <dt>
        Agenda Status:
      </dt>
      <dd>
        Coming soon
      </dd>
    </div>
  </div>
  <div
    className="employee-agenda-card-bottom"
  >
    <div
      className="button-container"
    >
      <div
        className="view-agenda-item-container"
      >
        <LinkButton
          className="view-agenda-item-button"
          toLink="/profile/ao/agendaitemhistory/"
        >
          View History
        </LinkButton>
      </div>
      <div
        className="create-ai-box-container"
      />
    </div>
  </div>
</BoxShadow>
`;
